public/app/plugins/datasource/loki/addToQuery.ts

```diff
@@ -227,3 +254,12 @@ function addParser(query: string, queryPartPositions: Position[], parser: string
 function labelExists(labels: QueryBuilderLabelFilter[], filter: QueryBuilderLabelFilter) {
   return labels.find((label) => label.label === filter.label && label.value === filter.value);
 }
+
+/**
+ * Return the last position based on "to" property
+ * @param positions
+ */
+function findLastPosition(positions: Position[]): Position {
+  const sorted = sortBy(positions, (position) => position.to);
```

Comment:
Since this is just about getting the element of `positions` where `position.to` is the highest, I think we could simply use a reducer:
```
return positions.reduce((prev, current) => (prev.to > current.to) ? prev : current);
```